src/components/todosItem.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import { useContext } from 'react';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './todosContext';
+import { ActiveContext, AllContext, CompletedContext } from './filterContext';
+import { ManageCheckboxContext } from './manageCheckboxContext';
+import classNames from 'classnames';
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+interface ItemProps {
+  item: Todo;
+}
+
+export const TodosItem: React.FC<ItemProps> = ({ item }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const { isAllSelected } = useContext(AllContext);
+  const { isActiveSelected } = useContext(ActiveContext);
+  const { isCompletedSelected } = useContext(CompletedContext);
+  const { setIsChecked } = useContext(ManageCheckboxContext);
+
+  if (todos.every(element => element.completed === true)) {
+    setIsChecked(true);
+  } else {
+    setIsChecked(false);
+  }
+
+  const changePersonalComplete = () => {
+    return todos.map(elem => {
+      if (elem.id === item.id) {
+        return {
+          ...elem,
+          completed: !item.completed,
+        };
+      }
+
+      return elem;
+    });
+  };
+
+  const toRender = () => {
+    if (isAllSelected === true) {
+      return true;
+    } else if (isActiveSelected && !item.completed) {
+      return true;
+    } else if (isCompletedSelected === true && item.completed === true) {
+      return true;
+    } else {
+      return;
+    }
+  };
+
+  const isTodoCompletedClass = classNames({
+    todo: true,
+    completed: item.completed,
+  });
+
+  return (
+    <>
+      {toRender() === true && (
```

Comment:
this check looks redundant, you always render the TodoItem

Reply:
No, there are fiters , and depends on it we decide which 'item' shoud be rendered or not